akka-http-core/src/main/scala/akka/http/engine/client/HostConnectionPoolSettings.scala

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.client
+
+import akka.event.LoggingAdapter
+import akka.http.model.HttpHeader
+import akka.io.Inet
+import com.typesafe.config.Config
+import scala.collection.immutable
+import scala.concurrent.duration.{ FiniteDuration, Duration }
+import akka.actor.ActorRefFactory
+import akka.http.util._
+
+final case class HostConnectionPoolSetup(host: String, port: Int, setup: ConnectionPoolSetup)
+
+final case class ConnectionPoolSetup(
+  encrypted: Boolean,
```

Comment:
I'm not quite sure yet of how exactly things will look. But something along these lines, yes.